### DISCUSSION TITLE
Handling inputs with 'type="number"' in typescript

### DISCUSSION THREAD

**Discussion by laurentaubin** [react-hook-form/react-hook-form#9376] [Q&A] [answered]
Hey, I've had some issues handling inputs with the `number` type in Typescript. When changing the input's value, the field is set to a string, instead of the expected number type. React hook form seems to infer the types correctly at compile time, but the type is wrong at runtime.

Is there any way to handle the input as a `number` in validation functions ?

Here's a simplified version of my code (using MUI's `TextField`)

```typescript
const someCustomValidationMethod = (value) => {
   // baseFormObject.field is typed as a number but is actually a string at runtime
   return baseFormObject.field < 10;
}

 <Controller
            name={`field`}
            defaultValue={0}
            control={control}
            rules={{ validate: someCustomValidationMethod, required: true }}
            render={({ field: { onChange, value }, fieldState: { error } }) => (
                <TextField
                  type="number"
                  InputProps={{ inputProps: { min: 0, max: 100 } }}
                  onChange={onChange}
                  value={value}
                />
              </Grid>
            )}
          />
```

As a quick fix, I can cast the value inside my validation function, but that's a bit ugly: 

```typescript
const someCustomValidationMethod = () => {
   return baseFormObject.field as unknown as number < 10;
}
```

Is there something I'm missing ?

**Comment by bluebill1049** [comment, marked as answer]
Native input will always return type as string by default unless invoke `valueAsNumber` at runtime.
